core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -506,16 +506,32 @@ protected InetSocketAddress getBroadcastRpcAddress(
     }
     // system.local for Cassandra >= 4.0
     Integer broadcastRpcPort = row.getInteger("rpc_port");
+    Integer broadcastRpcPortSsl;
     if (broadcastRpcPort == null || broadcastRpcPort == 0) {
       // system.peers_v2
       broadcastRpcPort = row.getInteger("native_port");
+      broadcastRpcPortSsl = row.getInteger("native_port_ssl");
+
       if (broadcastRpcPort == null || broadcastRpcPort == 0) {
-        // use the default port if no port information was found in the row;
-        // note that in rare situations, the default port might not be known, in which case we
-        // report zero, as advertised in the javadocs of Node and NodeInfo.
-        broadcastRpcPort = port == -1 ? 0 : port;
+        if (broadcastRpcPortSsl == null || broadcastRpcPortSsl == 0) {
+          broadcastRpcPort = port == -1 ? 0 : port;
+        } else {
+          broadcastRpcPort = broadcastRpcPortSsl;
+        }
+      } else {
+        if (broadcastRpcPortSsl != null && broadcastRpcPortSsl != 0) {
+          if (!broadcastRpcPortSsl.equals(broadcastRpcPort)) {
+            broadcastRpcPort = broadcastRpcPortSsl;
```

Comment:
We only want to perform this substitution if the driver is configured to use SSL.  If it isn't it's picture of the nodes should still use what's in native_port.